sagemaker-python-sdk/pytorch_horovod_mpi/pytorch_horovod_mpi.ipynb

```diff
@@ -0,0 +1,791 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Training PyTorch Models using Horovod with Open MPI"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Background\n",
+    "\n",
+    "Horovod is a distributed deep learning training framework that is used to speed up deep learning workloads. This notebook will show you how to train your PyTorch models using Horovod with Open MPI. Using the Sagemaker Python SDK, we will:\n",
+    "\n",
+    "* Upload a data set to S3\n",
+    "* Train a simple neural network with Horovod\n",
+    "* Deploy a model to a SageMaker endpoint\n",
+    "\n",
+    "You can learn more about Horovod and MPI at the following links:\n",
+    "\n",
+    "* https://horovod.readthedocs.io/en/stable/pytorch.html\n",
+    "* https://horovod.readthedocs.io/en/stable/mpirun.html \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "For this example, we'll use the MNIST data set. MNIST is a widely used dataset for handwritten digit classification. It consists of 70,000 labeled 28x28 pixel grayscale images of hand-written digits. The dataset is split into 60,000 training images and 10,000 test images. There are 10 classes (one for each of the 10 digits)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Downloading the Data\n",
+    "\n",
+    "The MNIST data set is easily accessible through the `torchvision.datasets` package. Note that even though we've set `train=false`, the code downloads both the training and testing data."
```

Comment:
> Note that even though we've set `train=false`, the code downloads both the training and testing data."

is this intentional/expected? seems to run contrary to what [the PyTorch docs](https://pytorch.org/docs/stable/torchvision/datasets.html#torchvision.datasets.MNIST) say

Reply:
If you set `download=True`, both the training and testing data is downloaded. When you set `train=False`, it means that you load the testing data (i.e `Dataset.__getitem__` returns data from the test set as opposed to from the train set).

I'm not sure how to concisely articulate that, so I'll just remove the line for now. I don't want to confuse any of the users